### PR TITLE
Add `mask.new()`, which can merge two masks

### PR DESCRIPTION
**Pull request by eriknw**
Largely based on https://github.com/python-graphblas/python-graphblas/issues/234#issuecomment-1132304594

This does not yet make masks "value-y".

Light benchmarking was done to choose "good enough" recipes, but better recipes may exist.  The code is structured in a way that a recipe for any combination of masks may be easily updated.  This will help us improve a specific recipe if it's ever important for a benchmark.

This includes `complement=True` option to return the _complement_ of merged masks.

There have been times when I've wanted to do `.new()` on a mask.

Hopefully the results are intuitive.  I think they are.

### CODE REVIEW

**Comment by eriknw** [comment]
Failing tests are b/c we need to update `python-suitesparse-graphblas`.  I can do this.

**Comment by eriknw** [comment]
More efficient recipes exist for some combos if we want the result to be used as a value mask--i.e., if it's allowed to have both True and False values, and not only True values.

I think this is less intuitive than only having True values.  But, we could support this in the future if there's a desire (and perhaps a performance benefit to the overall workload).  Heh, I guess we could add something like `value=False` or `structure=True` keyword argument to `.new`.

**Review by jim22k — APPROVED** [review]
Overall looks good

File: graphblas/mask.py [hunk 1]
```diff
@@ -34,6 +40,47 @@ def mask(self):
         )
         return self.parent
 
+    def new(self, dtype=None, *, complement=False, mask=None, name=None):
+        """Return a new object with True values determined by the mask.
+
+        The resulting object used as structural or value mask is equivalent to the original mask.
```
[the remark this answers is not in the document]
Comment (eriknw) [reply graphblas/mask.py:46]: Refers to `self`.  How can we make this clearer?  Basically `A(result.S) << expr` is the same as `A(self) << expr`

File: graphblas/mask.py [hunk 2]
```diff
@@ -105,3 +152,218 @@ def name(self):
...
+# Recipes to combine two masks.
+# Legend:
+#    A: any
+#    S: structural
+#    V: value
+#    CS: complemented structural
+#    CV: complemented value
+def _combine_S_S(m1, m2, dtype, name):
+    """S-S"""
+    return pair(one(m1.parent).new() & one(m2.parent).new()).new(dtype, name=name)
```
[the remark this answers is not in the document]
Comment (eriknw) [reply graphblas/mask.py]: Actually, good catch.  This was my bad.  Probably copy/pasted from using `pair(one(x) | one(y))`.  Updated.

**Pull request merged**